packages/strapi-plugin-versioning/src/services/Versioning.ts

```diff
@@ -23,33 +27,57 @@ const findUid = (ctx: Context): string =>
   ctx.params.model
     ? global.strapi.db.getModel(ctx.params.model).uid
     : Object.values(global.strapi.models).find(
-        (el) => el.collectionName == ctx.url.split("/")[1],
+        (el) => el.collectionName == ctx.url.split('/')[1],
       )!.uid;
 
 const saveDataInDB = async (
   data: Data,
   model: string,
   entity: Entity,
 ): Promise<void> => {
-  const knexQueryBuilder = global.strapi.connections.default("versions");
+  const knexQueryBuilder = global.strapi.connections.default('versions');
   knexQueryBuilder
     .insert({
       date: new Date().toISOString(),
       content_type: model,
-      content: JSON.stringify(data),
+      content: isObject(data) ? JSON.stringify(data) : data,
       entity_id: entity.id,
-      entity: JSON.stringify(entity),
+      entity: isObject(data) ? JSON.stringify(data) : data,
     })
-    .then();
+    .then(noop)
+    .catch(noop);
 };
 
 const getVersionsForAllConentTypes = async () => {
-  const knexQueryBuilder = global.strapi.connections.default("versions");
-  knexQueryBuilder.select().returning("*").toString();
-  return await knexQueryBuilder;
+  const knexQueryBuilder = global.strapi.connections.default('versions');
+  knexQueryBuilder.select().returning('*').toString();
+  return knexQueryBuilder;
+};
+
+const getVersionsForEntity = async (contentType: string, id: string) => {
+  const model = global.strapi.query(contentType);
+  const knexQueryBuilder = global.strapi.connections.default('versions');
+  knexQueryBuilder
+    .select();
+  if (model) {
+    const { model: { kind } } = model;
+    if (kind === 'singleType') {
+      const { id: singleTypeId } = await model.findOne({});
+      knexQueryBuilder
+        .where({ content_type: contentType, entity_id: singleTypeId });
```

Comment:
We need to check if the query returns actually an entry, I changed this block to:
```
const entry = await model.findOne({});
if (entry) {
  knexQueryBuilder.where({
    content_type: contentType,
    entity_id: entry.id,
  });
}
```

Reply:
good catch I will fix this in next commit :) thank you